pkgs/cronet_http/tool/prepare_for_embedded.dart

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+// ignore_for_file: lines_longer_than_80_chars
+
+import 'dart:io';
+
+import 'package:http/http.dart' as http;
+import 'package:xml/xml.dart';
+import 'package:yaml_edit/yaml_edit.dart';
+
+/// The cronet_http directory is used to produce two packages:
+/// - `cronet_http`, which uses the Google Play Services version of Cronet.
+/// - `cronet_http_embedded`, which embeds Cronet.
+///
+/// The default configuration of this code is to use the Google Play Services version of Cronet.
+///
+/// The script transforms the configuration into one that embeds Cronet by:
+/// 1. Modifying the Gradle build file to reference the embedded version of Cronet.
+/// 2. Modifying the *name* and *description* in `pubspec.yaml`.
+/// 3. Replaying `README.md` with `README_EMBEDDED.md`.
+///
+/// After running this script, `flutter pub publish` can be run to update package:cronet_http_embedded.
+///
+/// NOTE: This script modifies the above files in place.
+late final Directory _directory;
+
+const String _gmsDependencyName = 'com.google.android.gms:play-services-cronet';
```

Comment:
All of these type declarations are redundant, please remove them i.e. just write
const _gmsDependencyName = "...";

Reply:
Used to write with Flutter. :)